test/component_model_tests/climaatmos_standalone/atmos_driver.jl

```diff
@@ -1,12 +1,119 @@
+import ClimaComms
+import ArgParse
+import CUDA
+import ClimaCoupler
 import ClimaAtmos as CA
 import Random
 
 redirect_stderr(IOContext(stderr, :stacktrace_types_limited => Ref(false)))
 Random.seed!(1234)
 
-if !(@isdefined config)
-    config = CA.AtmosConfig()
+function argparse_settings()
+    s = ArgParse.ArgParseSettings()
+    ArgParse.@add_arg_table! s begin
+        "--config_file"
+        help = "A yaml file used to set the configuration of the coupled model"
+        arg_type = String
+        default = nothing
+    end
+    return s
 end
-simulation = CA.get_simulation(config)
-(; integrator) = simulation
+
+"""
+    show_memory_usage(comms_ctx, objects, io::Union{Base.TTY, IOBuffer} = stdout)
+
+Display the current memory footprint of the simulation, using an appropriate
+method based on the device being used.
+In the GPU case, show and return the memory usage of the GPU. Note that the
+return value in this case is a vector of unicode hex characters that must be
+converted to text to read (e.g. via a call`write(file, alloc_info)`).
+In the CPU case, show and return the memory footprint of the provided object(s) in GB.
+Note that these two cases provide different information, and should not be
+directly compared.
+# Arguments
+`comms_ctx`: the communication context being used to run the model
+`objects`: Dict mapping objects whose memory footprint is displayed in the CPU case to their names
+`io`: The input/output stream to use. Currently, `stdout` and `IOBuffer` types are supported.
+"""
+function show_memory_usage(comms_ctx, objects, io::Union{Base.TTY, IOContext, IOBuffer} = stdout)
```

Comment:
Is this the same as the one above? If so, could we add this to Utilities and use it from both drivers?

Reply:
good point!